### PR TITLE
Update compilation process lecture

### PR DESCRIPTION
**Pull request by ivankochin**
- Rename "compilation process" to "compilation stages"
- Move static and inline topics to link section
- Add some comments

### CODE REVIEW

File: 00_compilation_stages/00_01_compilation/00_01_01_extern/00_01_01_extern.cpp [hunk 1]
```diff
@@ -8,3 +8,5 @@ int main() {
     int local_a = global_a;
     return 0;
 }
+
+int global_a{};
```
Comment (rarutyun) [inline 00_compilation_stages/00_01_compilation/00_01_01_extern/00_01_01_extern.cpp]: Seems not relevant change

File: 00_compilation_stages/00_02_link/00_02_03_link_error/00_02_03_second_source.cpp [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#include "00_02_01_header.h"
+#include "00_02_03_header.h"
```
Comment (rarutyun) [inline 00_compilation_stages/00_02_link/00_02_03_link_error/00_02_03_second_source.cpp]: Where is _02 and so forth?

**Review by rarutyun — COMMENTED** [review]
Why were so many files renamed?

**Comment by ivankochin** [comment]
> Why were so many files renamed?

Because "compilation_process" folder was renamed to "compilation_stages"

**Pull request merged**